src/Types/PlutusData.purs

```diff
@@ -23,3 +36,40 @@ derive instance Generic PlutusData _
 
 instance Show PlutusData where
   show x = genericShow x
+
+-- On-chain PlutusData. Do we need this representation to separate when writing
+-- the Contract Monad?
+newtype BuiltinPlutusData = BuiltinPlutusData PlutusData
+
+derive newtype instance Eq BuiltinPlutusData
+derive newtype instance Ord BuiltinPlutusData
+derive instance Newtype BuiltinPlutusData _
+derive instance Generic BuiltinPlutusData _
+
+instance Show BuiltinPlutusData where
+  show = genericShow
+
+class ToPlutusData (a :: Type) where
+  toBuiltinPlutusData :: a -> BuiltinPlutusData
+
+-- FIX ME (create an issue unless someone notices simple solution to this in PR
+-- review)
+unitRedeemer :: RedeemerTag -> Redeemer
+unitRedeemer = undefined
+
+class FromPlutusData (a :: Type) where
+  -- | Convert a value from 'BuiltinPlutusData', returning 'Nothing' if this fails.
+  fromBuiltinPlutusData :: BuiltinPlutusData -> Maybe a
+
+newtype Datum = Datum BuiltinPlutusData
+
+derive newtype instance Eq Datum
+derive newtype instance Ord Datum
+derive instance Newtype Datum _
+derive instance Generic Datum _
+
+instance showDatum :: Show Datum where
+  show = genericShow
+
+-- Do we want a newtype?
+type DatumHash = DataHash
```

Comment:
I don't think we even need the type synonym, unless I'm missing something

Reply:
Yeah, it's probably fine without, I was following `type TxOutputRef = TransactionInput` to make it closer to Plutus when people "copy and paste" code over but can remove.

Reply:
Hmmmm. That's a good point